dataflux_pytorch/benchmark/checkpointing/simulated/multi_node_benchmark.py

```diff
@@ -49,56 +44,66 @@ def init_processes() -> int:
-    load_checkpoint_times = time_checkpoint_operation(
-        benchmark_strategy, distributed_state_dict, filepath, sample_size, 'load')
+        project=project,
+        path=filepath,
+    )
+
+    state_dict = dict()
+    for i in range(padding_size):
+        if i % world_size == rank:
+            state_dict[f'dummy_tensor_{i}'] = torch.randn(layer_size, 1000)
+    # Wait until the state_dict is populated properly accross all the nodes.
+    dist.barrier()
+
+    save_checkpoint_times = time_checkpoint_operation(benchmark_strategy,
+                                                      state_dict, filepath,
+                                                      sample_count, 'save',
+                                                      rank, world_size,
+                                                      padding_size, layer_size)
+
+    load_checkpoint_times = time_checkpoint_operation(benchmark_strategy,
+                                                      state_dict, filepath,
+                                                      sample_count, 'load',
+                                                      rank, world_size,
+                                                      padding_size, layer_size)
 
     if rank == 0:
-        print("######################")
-        print(
-            f"Time taken to save checkpoint: {statistics.mean(save_checkpoint_times):.4f} seconds")
-        print(
-            f"Time taken to load checkpoint: {statistics.mean(load_checkpoint_times):.4f} seconds")
-        total_distributed_size_bytes = sum(get_tensor_size_bytes(
-            tensor) for tensor in distributed_state_dict.values())
-        print(
-            f"Size of distributed tensors (rank {rank}): {format_size(total_distributed_size_bytes)}")
-        print(
-            f"Total size of all tensors (rank {rank}): {format_size(total_distributed_size_bytes * world_size)}")
+        print(f"Time taken to save checkpoint:\
+                {statistics.mean(save_checkpoint_times):.4f} seconds")
+        print(f"Time taken to load checkpoint:\
+                 {statistics.mean(load_checkpoint_times):.4f} seconds")
+
+        tensor_size_per_instance = 1000 * layer_size * state_dict[
+            f'dummy_tensor_0'].element_size()
+        tensors_per_rank = padding_size // world_size
+        total_size_bytes = total_tensor_count * tensor_size_per_instance * world_size
+        print(f"Size of distributed tensors (rank {rank}):\
+                 {format_size(total_tensor_count * tensor_size_per_instance)}")
+        print(f"Total size of all tensors:\
+                 {format_size(total_size_bytes)}")
         print("######################")
 
     cleanup()
 
 
 if __name__ == "__main__":
     world_size = int(os.getenv("WORLD_SIZE"))
-    model_size = int(os.getenv("NUM_LAYERS"))
+    layer_size = int(os.getenv("LAYER_SIZE"))
+    layer_size = int(os.getenv("LAYER_SIZE"))
```

Comment:
Duplicated code, is one of them supposed to be something else?

Reply:
Removed (messed up during the merge rebase)